etl/src/main/java/org/pih/hivmigration/etl/sql/EncounterMigrator.groovy

```diff
@@ -54,7 +54,8 @@ class EncounterMigrator extends SqlMigrator {
                     e.note_title,
                     e.response_to
                 from
-                    hiv_encounters e
+                    hiv_encounters e, hiv_demographics p 
+                    where e.patient_id = p.patient_id and (p.treatment_status is null or p.treatment_status != 'test')
```

Comment:
The hiv_demographics_real view takes care of this for you... ie, you just join on  "hiv_demographics_real" instead of hiv_demographics, as that is a view with all test patients excluded.

Reply:
From what I remember from a long time ago, joining on a view is much slower than joining on the real table. I think that Oracle made some improvements and it sounds like a Materialized View is much faster, but I do not think that hiv_demographics_real is a materialized view?

Reply:
Interesting point @cioan ...   Looks like the job had been taking about 150-160ish minutes to run currently.

Let's see what the time is after this addition and if it's significant look at ways to tweak.  (Since there's no join now, this would be just a comparison of no join vs the view join... but we can try to do more details tests if we find this adds a lot to the build).

fyi @mseaton

Reply:
It's certainly easy enough to test the performance of one query that joins against the view, and another that doesn't.  My personal assumption is that joining against a view is faster or the same, as the query engine knows what to expect up-front (the view has been created, even if non-matierialized) and Oracle could theoretically take indexing and other steps to optimize for it.

That said, the other approach is just to make sure we join on the hivmigration_patients or hivmigration_encounters table at the time of import into associated staging tables, since these will have already been created with the appropriate patients and encounters for migration.

We should treat those as foreign keys that need to be checked, even if those foreign key relationships don't actually exist in the staging tables.